mlir/test/Quantum/ScatterTest.mlir

```diff
@@ -117,13 +122,28 @@ func.func public @two_scatter(%arg0: tensor<3xf64>, %arg1: tensor<i64>) -> tenso
     // CHECK-NEXT:   return [[RESTENSOR1]] : tensor<f64>
 
 // CHECK: func.func public @two_scatter(%arg0: tensor<3xf64>, %arg1: tensor<i64>) -> tensor<3xf64>
-    // CHECK:   [[CST:%.+]] = arith.constant dense<3.000000e+00> : tensor<f64>
-    // CHECK:   [[CST0:%.+]] = arith.constant dense<2.000000e+00> : tensor<f64>
-    // CHECK:   [[INPUTS0:%.+]] = tensor.from_elements %extracted_3 : tensor<f64>
-    // CHECK:   [[CALL0:%.+]] = call @__catalyst_update_scatter[[NUMBER0]]([[INPUTS0]], [[CST]]) : (tensor<f64>, tensor<f64>) -> tensor<f64>
-    // CHECK:   [[INPUTS1:%.+]] = tensor.extract %arg0[%10] : tensor<3xf64>
-    // CHECK:   [[INPUTSEXTRACTED1:%.+]] = tensor.from_elements [[INPUTS1]] : tensor<f64>
-    // CHECK:   [[CALL1:%.+]] = call @__catalyst_update_scatter[[NUMBER1]]([[INPUTSEXTRACTED1]], [[CST0]]) : (tensor<f64>, tensor<f64>) -> tensor<f64>  
+    // CHECK:    [[IDX0:%.+]] = index.constant 0
+    // CHECK:    [[IDX1:%.+]] = index.constant 1
+    // CHECK:    [[CST0:%.+]] = arith.constant dense<2.000000e+00> : tensor<f64>
+    // CHECK:    [[CST1:%.+]] = arith.constant dense<3.000000e+00> : tensor<f64>
+
+    // CHECK:    [[SCF0:%.+]] = scf.for %arg2 = [[IDX0]] to [[IDX1]] step [[IDX1]] iter_args(%arg3 = %arg0) -> (tensor<3xf64>)
+    // CHECK:    [[INDEX:%.+]] = arith.index_cast %13 : i32 to index
+    // CHECK:    [[EXTRACTED:%.+]] = tensor.extract %arg3[[[INDEX]]] : tensor<3xf64>
+    // CHECK:    [[FROM:%.+]] = tensor.from_elements [[EXTRACTED]] : tensor<f64>
+    // CHECK:    [[CALL:%.+]] = func.call @__catalyst_update_scatter[[NUMBER0]]([[FROM]], [[CST1]]) : (tensor<f64>, tensor<f64>) -> tensor<f64>
+    // CHECK:    [[RES:%.+]] = tensor.extract [[CALL]][] : tensor<f64>
+    // CHECK:    [[INSERTED:%.+]] = tensor.insert [[RES]] into %arg3[[[INDEX]]] : tensor<3xf64>
+    // CHECK:    scf.yield [[INSERTED]] : tensor<3xf64>
+
+    // CHECK:    [[SCF1:%.+]] = scf.for %arg2 = [[IDX0]] to [[IDX1]] step [[IDX1]] iter_args(%arg3 = %arg0) -> (tensor<3xf64>)
+    // CHECK:    [[INDEX:%.+]] = arith.index_cast %13 : i32 to index
+    // CHECK:    [[EXTRACTED:%.+]] = tensor.extract %arg3[[[INDEX]]] : tensor<3xf64>
+    // CHECK:    [[FROM:%.+]] = tensor.from_elements [[EXTRACTED]] : tensor<f64>
+    // CHECK:    [[CALL:%.+]] = func.call @__catalyst_update_scatter[[NUMBER1]]([[FROM]], [[CST0]]) : (tensor<f64>, tensor<f64>) -> tensor<f64>
+    // CHECK:    [[RES:%.+]] = tensor.extract [[CALL]][] : tensor<f64>
+    // CHECK:    [[INSERTED:%.+]] = tensor.insert [[RES]] into %arg3[[[INDEX]]] : tensor<3xf64>
+    // CHECK:    scf.yield [[INSERTED]] : tensor<3xf64>
```

Comment:
How come there are two loops in this case? (mostly for my understanding)

Reply:
Because it is a case where we have two different scatter op with non similar update functions.